src/librustc/ty/mod.rs

```diff
@@ -1887,6 +1899,74 @@ impl<'a, 'gcx, 'tcx> AdtDef {
         debug!("sized_constraint_for_ty({:?}) = {:?}", ty, result);
         result
     }
+
+    fn dynsized_constraint_for_ty(&self,
+                                   tcx: TyCtxt<'a, 'tcx, 'tcx>,
+                                   ty: Ty<'tcx>)
+                                   -> Vec<Ty<'tcx>> {
+        let result = match ty.sty {
+            TyBool | TyChar | TyInt(..) | TyUint(..) | TyFloat(..) |
+            TyRawPtr(..) | TyRef(..) | TyFnDef(..) | TyFnPtr(_) |
+            TyStr | TyDynamic(..) | TySlice(_) | TyArray(..) |
+            TyTuple(..) | TyAdt(..) | TyClosure(..) | TyGenerator(..) |
+            TyNever => {
+                vec![]
+            }
+
+            TyForeign(..) | TyError => {
```

Comment:
I believe `TyError` probably wants to be considered `DynSized`

Reply:
Well, I guess we treat it the other way for sized-constraint. Probably doesn't matter since we'll just suppress the error we would have reported later on.